### ISSUE TITLE
is it npm ua-parser?

### ISSUE THREAD

**Issue by zowers** [ua-parser/uap-ref-impl#8]
is this the source of npm's ua-parser package?
https://www.npmjs.com/package/ua-parser

and why is the version https://github.com/ua-parser/uap-ref-impl/blob/master/package.json#L5
is lower then https://github.com/tobie/ua-parser/blob/master/package.json#L5?

**Comment by CedricConnes** [comment]
I agree, since the split of [tobie/ua-parser](https://github.com/tobie/ua-parser) to the [uaparser](https://github.com/ua-parser) project space, things became rather confusing for JavaScript users.

From what I understand:
 * [ua-parser/uap-core](https://github.com/ua-parser/uap-core) now wraps the up-to-date `regexes.yaml`, which contains the core definitions, shared by all language implementations.
 * [tobie/ua-parser](https://github.com/tobie/ua-parser) is the legacy repository, which holds its own outdated `regexes.yaml`. Apparently, this is what is currently published as [ua-parser](https://www.npmjs.com/package/ua-parser) on npm. 
 * [ua-parser/uap-ref-impl](https://github.com/ua-parser/uap-ref-impl) is supposed to be the official _JavaScript reference implementation_, as stated [here](https://github.com/tobie/ua-parser#ua-parser-has-moved). It sounds like a replacement for the deprecated [tobie/ua-parser](https://github.com/tobie/ua-parser), but it is not what happened yet (no documentation, no tests, low usage on npm, ...).

Any clarification is welcomed about this :smiley:.
I'm particularly interested on how JavaScript users are supposed to work with the up-to-date `regexes.yaml` now.

**Comment by zowers** [comment]
imo "JavaScript reference implementation" means something that is only for reference purposes, not for production use.

**Comment by josip** [comment]
:+1: this is rather confusing. There's now even a third npm package: https://www.npmjs.com/package/node-uap

**Comment by connesc** [comment]
I will end by using the [useragent](https://www.npmjs.com/package/useragent) package, which is widely used, and [internally fetches](https://github.com/3rd-Eden/useragent/blob/master/lib/update.js#L213) the up-to-date `regexes.yaml` from [ua-parser/uap-core](https://github.com/ua-parser/uap-core).

**Comment by vladikoff** [comment]
so confusing, took me like 5 minutes to find this cc @tobie

**Comment by tobie** [comment]
Unfortunately, I don't really have the time to fix things atm. Maybe later this year?

**Comment by vcarel** [comment]
Hi. Is there any update on this issue?

**Comment by tobie** [comment]
Unfortunately no. Would love to clean this mess up if someone was willing to sponsor the work. Can't afford the time away from client work otherwise. :(

**Comment by fedot** [comment]
@josip as a `node-uap` packager, I would love to see the [ua-parser/uap-core](https://github.com/ua-parser/uap-core) package published to npm registry. That way I could be using [ua-parser/uap-ref-impl](https://github.com/ua-parser/uap-ref-impl) directly (and contributing to it instead of rolling my wrap-up)

**Comment by Krinkle** [comment]
There is also [ua-parser-js](https://www.npmjs.com/package/ua-parser-js) (<https://github.com/faisalman/ua-parser-js>), which is gaining popularity and actively maintained.

**Comment by LinusU** [comment]
Spent time trying to figure out this as well 😄 very confusing...

In the end I decided to go with [`useragent`](https://npmjs.org/packages/useragent), it works great!
